### PR TITLE
move SWR cache from page global to components-global

### PR DESCRIPTION
**Pull request by doneel**

### CODE REVIEW

File: src/providers/LayerProvider/LayerProvider.tsx [hunk 1]
```diff
@@ -237,7 +235,6 @@ export const LayerProvider = ({
       params: { businessId },
     }),
     {
-      ...defaultSWRConfig,
```
Comment (doneel) [inline src/providers/LayerProvider/LayerProvider.tsx:240]: Removing the explicit setting of the default SWR config so it uses the single global config we define in the `SWRConfig`.

File: src/providers/LayerProvider/LayerProvider.tsx [hunk 2]
```diff
@@ -348,7 +345,7 @@ export const LayerProvider = ({
   const drawerContextData = useDrawer()
 
   return (
-    <SWRConfig value={defaultSWRConfig}>
+    <SWRConfig value={{ ...defaultSWRConfig, provider: () => new Map() }}>
```
Comment (doneel) [inline src/providers/LayerProvider/LayerProvider.tsx:348]: Every time we re-initialize this `<SWRConfig`, we'll reset the cache. That means if you unmount and remount the component, you don't need to actually call `expireCaches`, we'll just automatically refresh everything.

File: src/types/layer_context.ts [hunk 3]
```diff
@@ -33,7 +33,7 @@ export type LayerContextHelpers = {
   touch: (model: DataModel) => void
   read: (model: DataModel, cacheKey: string) => void
   syncTimestamps: Partial<Record<DataModel, number>>
-  readTimestamps: Partial<Record<DataModel, number>>
+  readTimestamps: Partial<Record<string, { t: number; m: DataModel }>>
```
Comment (doneel) [inline src/types/layer_context.ts:36]: This was previously wrong, just noticed it today while debugging.

**Review by AndrewAcomb — APPROVED** [review]

**Review by AndrewAcomb — APPROVED** [review]
Works well for me

**Pull request merged**